### PR TITLE
Add optional callZomeTransform

### PR DESCRIPTION
**Pull request by matthme**
Until recently it was possible to overwrite the zome call signing logic [like this](https://github.com/lightningrodlabs/we/blob/7bbdf403b6cffdedafbae66f4d231176cfe67ca3/utils/applet-iframe/src/index.ts#L337). But this doesn't seem to be possible anymore since there is now a `callZomeRequester` that's private and readonly. This PR allows to choose a custom zome call transform for use cases that the default CallZomeTransform cannot handle.

### CODE REVIEW

File: src/api/app/websocket.ts [hunk 1]
```diff
@@ -92,11 +100,15 @@ export class AppWebsocket implements AppClient {
   private constructor(
     client: WsClient,
     appInfo: AppInfo,
-    defaultTimeout?: number
+    defaultTimeout?: number,
+    callZomeTransform?: CallZomeTransform
```
Comment (ThetaSinner) [inline src/api/app/websocket.ts]: Which are users more likely to set? I'd say the `callZomeTransform` is probably more common than overriding the timeout so maybe that comes before the `defaultTimeout` in the params list?

Comment (ThetaSinner) [reply src/api/app/websocket.ts]: Probably not so important either way

**Review by ThetaSinner — APPROVED** [review]
Nice, looks good! One question that might not need looking at

File: src/api/app/websocket.ts (hunk 1 above)
Comment (matthme) [reply src/api/app/websocket.ts]: Good question, I was assuming the opposite but you might be right. Holochain has an internal timeout of 60 seconds is that right? So the only sensible reason to change the timeout here would be to make it shorter which I'm not sure many people would want to do.

File: src/api/app/websocket.ts [hunk 2]
```diff
@@ -92,11 +100,15 @@ export class AppWebsocket implements AppClient {
...
   ) {
     this.client = client;
     this.myPubKey = appInfo.agent_pub_key;
     this.defaultTimeout = defaultTimeout ?? DEFAULT_TIMEOUT;
+    this.callZomeTransform = callZomeTransform
```
Comment (jost-s) [inline src/api/app/websocket.ts]:
```suggestion
    this.callZomeTransform = callZomeTransform ?? defaultCallZomeTransform
```

**Review by jost-s — CHANGES_REQUESTED** [review]
Teeny change request

File: src/api/app/websocket.ts (hunk 2 above)
Comment (matthme) [reply src/api/app/websocket.ts]: done.

**Review by jost-s — APPROVED** [review]

**Pull request merged**